### PR TITLE
media-video/h264enc: re-add aac support

### PR DESCRIPTION
**Pull request by nephros**
This used to depend on media-sound/aacplusenc which has been removed
from Portage. The h264enc script actually only needs a compatible
aacplusenc executable, which is provided by media-libs/libaacplus.

  - EAPI=7
  - replace unavailable aacplusenc with libaacplus
  - patch the call to aacplusenc to match the new syntax
  - remove old versions
  - remove masking of aac USE flag

Bug: https://bugs.gentoo.org/630254
Signed-off-by: Peter Gantner <gentoo@nephros.org>

### CODE REVIEW

**Comment by juippis** [comment]
Is this still relevant? Could you rebase on top of fresh ::gentoo master and let's see what happens?

**Comment by nephros** [comment]
> Is this still relevant? Could you rebase on top of fresh ::gentoo master and let's see what happens?

I did a rebase locally and it went ahead with no issues. (But due to things entirely my own fault I had to do a force-push to github, no ifdea if that caused any damage.)

Any interest in this? I'll be happy to maintain this for myself in my local overlay if not.

File: media-video/h264enc/h264enc-10.4.7-r1.ebuild [hunk 1]
```diff
@@ -1,7 +1,7 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
```
Comment (juippis) [inline media-video/h264enc/h264enc-10.4.7-r1.ebuild:1]: 2020 please.

File: media-video/h264enc/h264enc-10.4.7-r1.ebuild [hunk 2]
```diff
@@ -20,7 +20,8 @@ RDEPEND="
 	sys-process/time
 	aac? (
 		media-libs/faac
-		media-sound/aacplusenc )
+		media-libs/libaacplus
+		)
```
Comment (juippis) [inline media-video/h264enc/h264enc-10.4.7-r1.ebuild:24]: One tab too many here.

File: media-video/h264enc/h264enc-10.4.7-r1.ebuild [hunk 3]
```diff
@@ -1,7 +1,7 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2019 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
 DESCRIPTION="Script to encode H.264/AVC/MPEG-4 Part 10 formats"
 HOMEPAGE="http://h264enc.sourceforge.net/"
```
Comment (juippis) [inline media-video/h264enc/h264enc-10.4.7-r1.ebuild:7]: `https://`

File: media-video/h264enc/h264enc-10.4.7-r1.ebuild [hunk 4]
```diff
@@ -30,6 +31,8 @@ RDEPEND="
 	vorbis? ( media-sound/vorbis-tools )"
 DEPEND=""
 
+PATCHES=( "${FILESDIR}"/libaacplusenc.patch )
```
Comment (juippis) [inline media-video/h264enc/h264enc-10.4.7-r1.ebuild:34]: Please rename the patch including the package version where this was introduced. Ie h264enc-10.4.1-libaacplusenc.patch

**Comment by juippis** [comment]
Actually now that I viewed the whole ebuild (sorry Github doesn't show it from here!) the whole ebuild needs a rewrite pretty much. 

We need to drop the USE flags and use `optfeature` from *eutils.eclass* while always applying the patch. Do you want to do it or should I?

https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

**Comment by nephros** [comment]
> Actually now that I viewed the whole ebuild (sorry Github doesn't show it from here!) the whole ebuild needs a rewrite pretty much.
> 
> We need to drop the USE flags and use `optfeature` from _eutils.eclass_ while always applying the patch. Do you want to do it or should I?
> 
> https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

I have not used that feature yet, so I'm not sure exactly what you intend.
Feel free to write what ever ebuild you like of course - but if you don't have the time or mood for it I'll be happy to have a go.

**Pull request closed without merging**